### PR TITLE
Start to implement/stub BSD:U and SFDNSRES services

### PR DESCRIPTION
**Pull request by flerovii**
I started to implement the ```bsd:u``` and ```sfdnsres``` services in order to get the BSD test further.

### CODE REVIEW

File: src/core/hle/service/bsd/bsd_u.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+void BSD_U::Socket(Kernel::HLERequestContext& ctx) {
+    IPC::RequestParser rp{ctx};
+
+    u32 domain = rp.Pop<u32>();
+    u32 type = rp.Pop<u32>();
+    u32 protocol = rp.Pop<u32>();
+
+    LOG_WARNING(Service, "(STUBBED) called domain=%d type=%d protocol=%d", domain, type, protocol);
```
Comment (lioncash) [inline src/core/hle/service/bsd/bsd_u.cpp]: These should be `%u`

Comment (flerovii) [reply src/core/hle/service/bsd/bsd_u.cpp]: fixed

File: src/core/hle/service/bsd/bsd_u.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+BSD_U::BSD_U() : ServiceFramework("bsd:u") {
+    static const FunctionInfo functions[] = {
+        {0, &BSD_U::RegisterClient, "RegisterClient"},
+        {2, &BSD_U::Socket, "Socket"}
+    };
+    RegisterHandlers(functions);
+}
+
+} // namespace BSD
+} // namespace Service
```
Comment (bunnei) [inline src/core/hle/service/bsd/bsd_u.cpp]: nit: Should be a new line at the end of the file (I think clang-format should catch this)

File: src/core/hle/service/sfdnsres/sfdnsres.cpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Copyright 2018 yuzu emulator team
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#include "core/hle/service/sfdnsres/sfdnsres.h"
+
+namespace Service {
+namespace Sfdnsres {
+
+Sfdnsres::Sfdnsres() : ServiceFramework("sfdnsres") {
```
Comment (bunnei) [inline src/core/hle/service/sfdnsres/sfdnsres.cpp]: Remove, and just move to the header, i.e.
`Sfdnsres() : ServiceFramework("sfdnsres") {}`

File: src/core/hle/service/sfdnsres/sfdnsres.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    Sfdnsres();
+    ~Sfdnsres() = default;
+
+private:
+};
+
+void InstallInterfaces(SM::ServiceManager& service_manager);
+
+} // namespace Sfdnsres
+} // namespace Service
```
Comment (bunnei) [inline src/core/hle/service/sfdnsres/sfdnsres.h]: Missing a new line at the end of the file

**Comment by bunnei** [comment]
BSD:U and SFDNSRES are both "sockets" code. Can you put these under a common "sockets" directory in the service/ folder, and have them both under the `Sockets` namespace? Furthermore, you can add sockets.cpp/h, with a single `InstallInterfaces` function that initializes both services

File: src/core/hle/service/sfdnsres/sfdnsres.h [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+#pragma once
+
+#include "core/hle/kernel/hle_ipc.h"
+#include "core/hle/service/service.h"
+
+namespace Service {
+namespace Sfdnsres {
+
+class Sfdnsres final : public ServiceFramework<Sfdnsres> {
```
Comment (bunnei) [inline src/core/hle/service/sfdnsres/sfdnsres.h]: nit: use uppercase for this class name, e.g. `SFDNSRES`, to be consistent with the other services

**Comment by flerovii** [comment]
All requested changes should be included in the last commit :)

File: src/core/hle/service/service.cpp [hunk 6]
```diff
@@ -176,6 +177,7 @@ void Init() {
     PCTL::InstallInterfaces(*SM::g_service_manager);
     Time::InstallInterfaces(*SM::g_service_manager);
     VI::InstallInterfaces(*SM::g_service_manager);
+    Sockets::InstallInterfaces(*SM::g_service_manager);
```
Comment (bunnei) [inline src/core/hle/service/service.cpp]: Can you move this below `PCTL`? These are installed in alphabetical order. Thansk!

Comment (flerovii) [reply src/core/hle/service/service.cpp]: Done

**Comment by bunnei** [comment]
Unfortunately, the formatting is now broken - can you please install and run clang-format?

**Comment by Hexagon12** [comment]
Please fix the conflict, AppVeyor doesn't like that.

**Comment by flerovii** [comment]
Weird. https://github.com/yuzu-emu/yuzu/pull/78/commits/9964dcedc71f56cb802dd2f7fa13526fa838371d _should_ fix the conflict :|

**Comment by MerryMage** [comment]
Instead of merging master into your branch, one would preferably rebased their branch on top of master. :p

merge vs rebase:

screen shot 2018-01-18 at 13 53 14
screen shot 2018-01-18 at 13 53 56

**Comment by flerovii** [comment]
thanks, merrymage :p that worked i guess

**Comment by bunnei** [comment]
👍 lgtm ty

**Pull request merged**